### PR TITLE
created fewnick tree

### PR DESCRIPTION
**Pull request by shsarv**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@shsarv modular division should be in the maths section not in the blockchain section

File: Python/README.md [hunk 1]
```diff
@@ -10,11 +10,14 @@
 # Codes in the Python language
 
 ## Data Structures
+
```
Comment (Kajol-Kumari) [inline Python/README.md:63]: Please remove this unnecessary change @shsarv

File: Python/README.md [hunk 2]
```diff
@@ -73,7 +77,7 @@
 _add list here_
 
 ## Blockchain
-_add list here_
```
Comment (Kajol-Kumari) [inline Python/README.md]: Please undo the changes done here @shsarv

File: Python/blockchain/README.md [hunk 3]
```diff
@@ -1 +1,3 @@
+### Blockchain
+
```
Comment (Kajol-Kumari) [inline Python/blockchain/README.md]: please remove these unnecessary changes @shsarv

File: Python/ds/Tree/fenwick_tree.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        ret = 0
+        while i > 0:
+            ret += self.ft[i]
+            i -= i & (-i)
+        return ret
+
+
+if __name__ == "__main__":
+    f = FenwickTree(100)
+    f.update(1, 20)
```
Comment (Kajol-Kumari) [inline Python/ds/Tree/fenwick_tree.py]: Can you please take these input from user? Hardcoded values don't seems good in CP @shsarv

File: Python/math/modular_division.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+# Extended Euclid's Algorithm : If d divides a and b and d = a*x + b*y for integers x and y, then d = gcd(a,b)
+
+
+def extended_gcd(a, b):
+    """
+    >>> extended_gcd(10, 6)
+    (2, -1, 2)
+    >>> extended_gcd(7, 5)
+    (1, -2, 3)
+   ** extended_gcd function is used when d = gcd(a,b) is required in output
```
Comment (Kajol-Kumari) [inline Python/math/modular_division.py]: please remove the extra space here before `**`

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @shsarv when I ran you program  I am getting 1 testcase failing for each of extended_gcd,      greatest_common_divisor and modular_division.

Also please add an output of your first program below your code section in commented form in your first program.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
Please try dynamic input instead of hardcoding

File: Python/ds/fenwick_tree.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+
+if __name__ == "__main__":
+    n = int(input("size of fenwick tree    "))
+    f = FenwickTree(n)
+    for i in range(int(input("how many data you want to add   "))):
+        index, value = map(int, input("enter index and value    ").split(","))
+        f.update(index, value)
+    print(f.query(int(input("enter index to get query cumulative data   "))))
+    print(f.query(2))
```
Comment (Kajol-Kumari) [inline Python/ds/fenwick_tree.py]: Hey @shsarv I think we can remove these hardcoded indexes here from line 32 to 35 as we are now taking input from the user.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @shsarv please incorporate the suggested changes, rest looks good to me.

File: Python/ds/fenwick_tree.py (hunk 6 above)
Comment (shsarv) [reply Python/ds/fenwick_tree.py]: yeah sure.

**Review by Kajol-Kumari — COMMENTED** [review]
Hey @shsarv Thank you for incorporating the changes. Please add the sample input-output at the end of the code section in commented form [ I think you missed it from my previous comment :) ]. Rest is good to go :)

**Comment by shsarv** [comment]
@Kajol-Kumari  may you please tell me that in which file you want me to add the input -output ? because its already there in fenwick tree program. where as in modular division , i already added expected output for an given input for the test mode. so is that you want me to add the output obtained by running the test mode? or should i need to to take input from the user?

**Comment by Abhijit2505** [comment]
Please try to remove merge conflicts.
